test/file.ts

```diff
@@ -3205,6 +3206,60 @@ describe('File', () => {
     });
   });
 
+  describe('isPublic', () => {
+    const sandbox = sinon.createSandbox();
+
+    afterEach(() => sandbox.restore());
+
+    it('should execute callback with `true` in response', done => {
+      sandbox.stub(gaxios, 'request').resolves();
+      file.isPublic((err: gaxios.GaxiosError, resp: boolean) => {
+        assert.ifError(err);
+        assert.strictEqual(resp, true);
+        done();
+      });
+    });
+
+    it('should execute callback with `false` in response', done => {
+      sandbox.stub(gaxios, 'request').rejects({code: '403'});
+      file.isPublic((err: gaxios.GaxiosError, resp: boolean) => {
+        assert.ifError(err);
+        assert.strictEqual(resp, false);
+        done();
+      });
+    });
+
+    it('should propagate non-403 errors to user', done => {
+      const error = {code: '400'};
+      sandbox.stub(gaxios, 'request').rejects(error as gaxios.GaxiosError);
+      file.isPublic((err: gaxios.GaxiosError) => {
+        assert.strictEqual(err, error);
+        done();
+      });
+    });
+
+    it('should correctly send a HEAD request', done => {
+      const spy = sandbox.spy(gaxios, 'request');
+      file.isPublic((err: gaxios.GaxiosError) => {
+        assert.ifError(err);
+        assert.strictEqual(spy.calledWithMatch({method: 'HEAD'}), true);
+        done();
+      });
+    });
+
+    it('should correctly format URL in the request', done => {
+      const expecterURL = `http://${
+        BUCKET.name
+      }.storage.googleapis.com/${encodeURIComponent(file.name)}`;
```

Comment:
Could you check that the name is encoded by throwing a special character in there?

Reply:
good point, done